packages/teams-js/src/private/externalAppCardActions.ts

```diff
@@ -0,0 +1,169 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { FrameContexts } from '../public';
+import { errorNotSupportedOnPlatform } from '../public/constants';
+import { runtime } from '../public/runtime';
+
+/**
+ * @hidden
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export namespace externalAppCardActions {
+  /**
+   * @hidden
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export enum ActionOpenUrlType {
+    DeepLinkOther,
+    DeepLinkStageView,
+    DeepLinkDialog,
+    GenericUrl,
+  }
+
+  /**
+   * @hidden
+   * Error that can be thrown from IExternalAppCardActionService.handleActionOpenUrl
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface ActionOpenUrlError {
+    errorCode: ActionOpenUrlErrorCode;
+    message?: string;
+  }
+
+  /**
+   * @hidden
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export enum ActionOpenUrlErrorCode {
+    INVALID_LINK = 'INVALID_LINK', // Deep link is invalid
+    NOT_SUPPORTED = 'NOT_SUPPORTED', // Deep link is not supported
+    INTERNAL_ERROR = 'INTERNAL_ERROR', // Generic error
+  }
+
+  /**
+   * @hidden
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface IAdaptiveCardActionSubmit {
+    id: string;
+    data: Record<string, unknown>;
+  }
+
+  /**
+   * @hidden
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface ICardActionsConfig {
+    enableImback: boolean;
+    enableInvoke: boolean;
+    enableDialog: boolean;
+    enableStageView: boolean;
+    enableSignIn: boolean;
+    enableO365Submit: boolean;
+  }
+
+  /**
+   *
+   * @hidden
+   * Error that can be thrown from IExternalAppCardActionService.handleActionSubmit
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface ActionSubmitError {
+    errorCode: ActionSubmitErrorCode;
+    message?: string;
+  }
+
+  /**
+   * @hidden
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export enum ActionSubmitErrorCode {
+    INTERNAL_ERROR = 'INTERNAL_ERROR', // Generic error
+  }
+  /**
+   * @beta
+   * @hidden
+   * Delegates an Adaptive Card Action.Submit request to the host for the application with the provided app ID
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param appId ID of the application the request is intended for
+   * @param actionSubmitPayload The Adaptive Card Action.Submit payload
+   * @param cardActionsConfig The card actions configuration. This indicates which subtypes should be handled by this API
+   * @returns Promise that resolves when the request is completed and rejects with ActionSubmitError if the request fails
+   */
+  export function processActionSubmit(
+    appId: string,
+    actionSubmitPayload: IAdaptiveCardActionSubmit,
+    cardActionsConfig?: ICardActionsConfig,
```

Comment:
>? [](http://example.com/codeflow?start=21&length=1)

If this parameter isn't specified, what is the default config used?

Reply:
@ashishsingh2105 @susanshen96 do you know what this would be?

Reply:
@ashishsingh2105 will these ever not be passed? Should this be a required parameter?

Reply:
In the Teams codebase, it looks like the cardActionsConfig can be undefined, which means that there is no default config supplied if the action is not sent with one. @ashishsingh2105 - correct me if I'm wrong.

Reply:
Additionally - in the Teams codebase, I'm noticing that we define it as `cardActionConfig` (without the additional s). @ashishsingh2105 - does the naming make any difference?